src/components/submission/list.vue

```diff
@@ -485,27 +485,44 @@ export default {
 }
 
 #submission-list-actions {
-  align-items: baseline;
+  align-items: center;
   display: flex;
   flex-wrap: wrap-reverse;
+  // This is 10px of space between actions on the row and also 10px between
+  // rows if the actions start wrapping.
+  gap: 10px;
+  margin-bottom: 25px;
+
+  .form-inline {
+    margin-bottom: 0;
+    padding-bottom: 0;
+  }
```

Comment:
Different children of `#submission-list-actions` had different `margin-bottom` and `padding-bottom`, which may have been partly responsible for the vertical misalignment. Here, we remove the space below the `.form-inline`. Below, we remove the space between the download button. The latter spacing is now handled by the flexbox gap.